cdn/signedcookies/sign_cookie.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Command signedurls creates a signed URL for a Cloud CDN endpoint with the
+// given key.
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// [START example]
+
+// SignCookie creates a signed cookie for an endpoint served by Cloud CDN.
+//
+// - urlPrefix must start with "https://" and should include the path prefix
+// for which the cookie will authorize access to.
+// - key should be in raw form (not base64url-encoded) which is
+// 16-bytes long.
+// - keyName must match a key added to the backend service or bucket.
+func SignCookie(urlPrefix, keyName string, key []byte, expiration time.Time) (string, error) {
```

Comment:
A little different when this is a quickstart versus a standalone sample. But, we generally try to minimize args for standalone samples. We also give sample values inside the function body to make it easy to copy out.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#function-arguments-for-snippets